lib/ansible/cli/galaxy.py

```diff
@@ -1508,7 +1508,7 @@ def execute_list_collection(self, artifacts_manager=None):
         """
 
         output_format = context.CLIARGS['output_format']
-        collections_search_paths = set(context.CLIARGS['collections_path'])
+        collections_search_paths = set([C.COLLECTIONS_PATHS[0]])
```

Comment:
I don't think this is right. You are removing the ability for the collection path to be set on the CLI, in favor of only accepting the configuration value.

Reply:
You are right, and it did bring new errors  (but different ones!) ... this change does seem to at least fix the error about not finding the (default) collection path [see here](https://dev.azure.com/ansible/ansible/_build/results?buildId=38621&view=logs&j=05630f17-7677-51af-4b88-11118adc6a44&t=b3edd11b-a13a-5494-96a8-2c9d98447f6f&l=2505).
So now I need to find a way to have the `context.CLIARGS` handle default parameters that contain template values.

Reply:
Yesterday I responded in IRC, that the problem was due to using `C.config.get_config_value('CONFIG_NAME)` elsewhere in the codebase, and that they would need to be switched to use `C.CONFIG_NAME`.  Only uses of `C.CONFIG_NAME` have templating applied to them, and use of `C.config.get_config_value()` does not.

This is the specific line I am referring to:

https://github.com/ansible/ansible/blob/7c81a652c0af1831bb7e0afd0bb23c43e68a40bf/lib/ansible/plugins/loader.py#L1148

Reply:
That worked perfectly.

Except for the slight wrinkle where my (overeager?) change of a similar construct here: https://github.com/ansible/ansible/pull/77322/commits/a40927794960fa02dcd26a01010d74f816a30108#diff-fe8d3e13b2884db3e502001befe5ca81b0b203fe2cec9b31e2ee9b0f14a8e835R273 causes errors in some galaxy cache handling unit tests.